virttest/qemu_vm.py

```diff
@@ -3790,62 +3790,74 @@ def reboot(self, session=None, method="shell", nic_index=0,
         :param serial: Serial login or not (default is False).
         :return: A new shell session object.
         """
-        def __reboot(session):
+
+        def __get_session(nic_index=0, serial=False):
+            """
+            Get connection to VM
+            """
+            if serial:
+                return self.serial_login()
+            return self.login(nic_index=nic_index)
+
+        def __go_down(session, nic_index, serial):
             """
             Check guest rebooting.
             """
+            timeout = self.CLOSE_SESSION_TIMEOUT
+            try:
+                if serial:
+                    patterns = [r".*[Rr]ebooting.*", r".*[Rr]estarting system.*",
+                                r".*[Mm]achine restart.*", r".*Linux version.*"]
+                    if session.read_until_any_line_matches(
+                            patterns, timeout=timeout):
+                        session.close()
+                        return True
+                else:
+                    session = __get_session(nic_index, serial)
+                    session.close()
+                    return False
```

Comment:
if session is not a serial session and it is still alive, that mean guest still running not begin to reboot,  so go_down should be False.